api/src/main/java/org/openmrs/module/sync2/api/impl/SyncPushServiceImpl.java

```diff
@@ -49,9 +52,14 @@ public AuditMessage readDataAndPushToParent(String category, Map<String, String>
         auditMessage.setAction(action);
 
         try {
-            Object data = action.equals(ACTION_VOIDED) ? uuid :
-                    syncClient.pullData(category, clientName, localPull, CHILD);
-            syncClient.pushData(data, clientName, parentPush, action, PARENT);
+            Object localObj = action.equals(ACTION_VOIDED) ? uuid : syncClient.pullData(category, clientName, localPull, CHILD);
+            Object parentObj = syncClient.pullData(category, clientName, parentPull, PARENT);
+            pulledObjectExist = localObj != null ?
+                    compareLocalAndPulled(clientName, category, localObj, parentObj) : true;
```

Comment:
Are you sure about second part of this condition? Should pulledObjectExist equal true if localObj == null?